src/Controller/Hello.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class Hello
+{
+    use ControllerTrait;
+
+    /**
+     * @Route("/", name="homepage")
+     */
+    public function __invoke(Request $request): Response
```

Comment:
@nicolas-grekas You're totally right, I'm more talking about _"Preventing encouragement of coupling"_ rather than _real_ coupling.

@yceruto You're right, and as @jvasseur says, in MVC, a controller is supposed to get the user action, maybe use model and view layers, and return a result of the user actions. In Symfony, a controller is a class that can do lots of actions. It's a bit counter-intuitive.

@mickaelandrieu This is the purpose of the ADR (Action-Domain-Responder) pattern (a MVC refinement attempt) which has been proposed by @dunglas for a long time now for Symfony (for example with his [DunglasActionBundle](https://github.com/dunglas/DunglasActionBundle))
I find this pattern better than MVC because it's less fuzzy about how to implement actions, more flexible for the Domain part and a Responder could be anything, whereas MVC talks about views (but in APIs you don't technically have any View layer, for example).
_In fine_, Symfony won't be refined with only-invokable controllers (this would be a HUGE BC break), but encouraging people to more decouple their business logic is better for reusability and organizing your different classes.